### ISSUE TITLE
The error after model.fix_symbolic()

### ISSUE THREAD

**Issue by GuoMinghui07** [KindXiaoming/pykan#349]
I encountered a RuntimeError while performing symbolic regression. The error message indicates that one of the arguments has an illegal value. Below is the detailed error message and my code.

Here is the code I used for symbolic regression: 
`from kan import *
f = lambda x: torch.cosh(3*x[:, [0]]**2 + 2*x[:, [0]] + x[:, [1]])
dataset = create_dataset(f, n_var=2, ranges=[-1,1], train_num=1000)
model = MultKAN(width=[2,1,1], base_fun='identity', auto_save=True,seed=4)
model.fit(dataset, steps=100);
model.get_act(dataset)
model.plot()`

<img width="701" alt="截屏2024-07-17 11 47 52" src="https://github.com/user-attachments/assets/b23f702f-447f-4efa-b0e7-2f0107398d44">

I found the result of **fixing (1,0,0) with exp, r2=0.9992531538009644, c=2** was incorrect then I fixed it by using:
`model.unfix_symbolic(1,0,0)
model.fix_symbolic(1,0,0,'cosh')
model.fit(dataset, steps=50);`

and there was an error  similar to #287

<img width="952" alt="截屏2024-07-17 11 53 33" src="https://github.com/user-attachments/assets/351da065-6004-4206-a771-e4aa1cfbc1b1">

**Comment by HaHaTimGao** [comment]
I met the same problem.

**Comment by KindXiaoming** [comment]
maybe try model.fit(..., opt='Adam', lr=1e-3)?

**Comment by GuoMinghui07** [comment]
Thank you, it should be the gradient explosion causing the problem, but sometimes this issue arises after using fix_symbolic(), even though the R2 of the symbolic activation function is very high. I'm not quite sure about the reason:(

**Comment by KindXiaoming** [comment]
`model.fit(singularity_avoiding=True)` may also help
